folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -214,6 +214,7 @@ private DefaultRawMemcacheClient(final HostAndPort address,
       SetRequest setRequest = (SetRequest) request;
       byte[] value = setRequest.getValue();
       if (value.length > maxSetLength) {
+        pendingCounter.decrementAndGet();
```

Comment:
Good catch. Seems to me like this length check should be done before incrementing the pending counter even, avoiding the need for this decrement.